analysis/3_coregister-and-source-recon-bulk.py

```diff
@@ -0,0 +1,240 @@
+"""Coregistration and source reconstruction 
+Created AXL 28/10/24
+
+The script should be first run for all subjects. Then for subjects
+whose coregistration looked a bit off we re-run this script just for that
+particular subject.
+
+This does not use OSL batch utils/config-file setup.
+"""
+
+############## ------- Import packages and define functions needed ---------- ############
+
+import os
+import os.path as op
+import osl
+from pprint import pprint
+from osl import utils
+from osl import source_recon
+import numpy as np
+import coinsmeg_data as coinsmeg
+from IPython.display import HTML, display
+import mne
+from glob import glob
+import pathlib
+
+# functions
+def copy_polhemus_files(recon_dir, subject, preproc_file, smri_file, logger):
+    polhemus_headshape = np.loadtxt(op.join(polhemus_dir, 'polhemus_headshape.txt'))
+    polhemus_nasion = np.loadtxt(op.join(polhemus_dir, 'polhemus_nasion.txt'))
+    polhemus_rpa = np.loadtxt(op.join(polhemus_dir, 'polhemus_rpa.txt'))
+    polhemus_lpa = np.loadtxt(op.join(polhemus_dir, 'polhemus_lpa.txt'))
+    
+    #  Get coreg filenames
+    filenames = source_recon.rhino.get_coreg_filenames(recon_dir, subject)
+
+    # Save
+    np.savetxt(filenames["polhemus_nasion_file"], polhemus_nasion)
+    np.savetxt(filenames["polhemus_rpa_file"], polhemus_rpa)
+    np.savetxt(filenames["polhemus_lpa_file"], polhemus_lpa)
+    np.savetxt(filenames["polhemus_headshape_file"], polhemus_headshape)
+
+############## ------- Directories ---------- ############
+
+# Directories
+DATA_DIR = coinsmeg.RAW_DIR # this is the same as BASE_DIR as raw data is stored in the base directory
```

Comment:
replace here (and line 52) with data_dir rather than DATA_DIR, to reflect that this is local to this script rather than externally defined in coinsmeg.XX_DIR